packages/flutter/lib/src/material/chip_theme.dart

```diff
@@ -0,0 +1,425 @@
+  ///
+  /// Defaults to the ambient [ThemeData.chipTheme] if there is no
+  /// [ChipTheme] in the given build context.
+  ///
+  /// ## Sample code
+  ///
+  /// ```dart
+  /// class Spaceship extends StatelessWidget {
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return new ChipTheme(
+  ///       data: ChipTheme.of(context).copyWith(backgroundColor: const Color(0xff804040)),
+  ///       child: new ActionChip(
+  ///         label: const Text('Launch'),
+  ///         onPressed: () { print('We have liftoff!'); },
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// See also:
+  ///
+  ///  * [ChipThemeData], which describes the actual configuration of a chip
+  ///    theme.
+  static ChipThemeData of(BuildContext context) {
+    final ChipTheme inheritedTheme = context.inheritFromWidgetOfExactType(ChipTheme);
+    return inheritedTheme != null ? inheritedTheme.data : Theme.of(context).chipTheme;
+  }
+
+  @override
+  bool updateShouldNotify(ChipTheme oldWidget) => data != oldWidget.data;
+}
+
+/// Holds the color, shape, and typography values for a material design chip
+/// theme.
+///
+/// Use this class to configure a [ChipTheme] widget, or to set the
+/// [ThemeData.chipTheme] for a [Theme] widget.
+///
+/// To obtain the current ambient chip theme, use [ChipTheme.of].
+///
+/// The parts of a chip are:
+///
+///  * The "avatar", which is a widget that appears at the beginning of the
+///    chip. This is typically a [CircleAvatar] widget.
+///  * The "label", which is the widget displayed in the center of the chip.
+///    Typically this is a [Text] widget.
+///  * The "delete icon", which is a widget that appears at the end of the chip.
+///  * The chip is disabled when it is not accepting user input. Only some chips
+///    have a disabled state (i.e. [InputChip], [ChoiceChip], [FilterChip]).
+///
+/// See also:
+///
+///  * [Chip], a chip that displays information and can be deleted.
+///  * [InputChip], a chip that represents a complex piece of information, such
+///    as an entity (person, place, or thing) or conversational text, in a
+///    compact form.
+///  * [ChoiceChip], allows a single selection from a set of options. Choice
+///    chips contain related descriptive text or categories.
+///  * [FilterChip], uses tags or descriptive words as a way to filter content.
+///  * [ActionChip], represents an action related to primary content.
+///  * [CircleAvatar], which shows images or initials of entities.
+///  * [Wrap], A widget that displays its children in multiple horizontal or
+///    vertical runs.
+///  * [ChipTheme] widget, which can override the chip theme of its
+///    children.
+///  * [Theme] widget, which performs a similar function to [ChipTheme],
+///    but for overall themes.
+///  * [ThemeData], which has a default [ChipThemeData].
+class ChipThemeData extends Diagnosticable {
+  /// Create a [ChipThemeData] given a set of exact values. All the values
```

Comment:
We usually put explanatory text and examples in the class dart doc and leave the more prosaic stuff for the constructor's doc.

Reply:
OK, I moved the example up to the class dart doc.